src/Orchard/Data/SessionLocator.cs

```diff
@@ -71,7 +71,7 @@ public class TransactionManager : ITransactionManager, IDisposable {
 
         public void Cancel() {
             // IsActive is true if the transaction hasn't been committed or rolled back
-            if (_session != null && _session.Transaction.IsActive) {
+            if (_session != null && _session.Transaction != null && _session.Transaction.IsActive) {
```

Comment:
This is not correct. If the goal is to prevent `_session.Transaction` from being called if it's null, then this needs to be split in two `if` statements as the `DisposeSession` would no be called, even if `_session` itself is not null.